runtime/interpreter/interpreter.go

```diff
@@ -2253,7 +2234,8 @@ func (interpreter *Interpreter) declareInterface(
 			if nestedCompositeDeclaration.Kind() == common.CompositeKindEvent {
 				interpreter.declareNonEnumCompositeValue(nestedCompositeDeclaration, lexicalScope)
 			} else {
-				interpreter.declareTypeRequirement(nestedCompositeDeclaration, lexicalScope)
+				// this should be statically prevented in the checker
```

Comment:
Is this a TODO, or does it mean that the PR already changes the checker to reject this case? (Maybe change tense to avoid confusion)